src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/RuntimeBinder.cs

```diff
@@ -705,11 +705,8 @@ private Expr CreateArgumentEXPR(ArgumentObject argument, LocalVariableSymbol loc
             // as well so that overload resolution can find them.
             if (callingType.IsWindowsRuntimeType())
             {
-                TypeArray collectioniFaces = callingType.GetWinRTCollectionIfacesAll(SymbolLoader);
```

Comment:
I wonder why TypeArray even exists. It seems to be a trivial wrapper over `CType[]`
Its comments say `// Encapsulates a type list, including its size.` I guess that could be useful in C++

SInce array knows its size, the only value-add is the `AssertValid()`, which could be an extension method.

I wonder - why not just remove the type?